### PR TITLE
Add test functions to test new unify function

### PR DESCRIPTION
**Pull request by nordme**
Adds test functions to test unify_bad_channels.

### CODE REVIEW

**Comment by nordme** [comment]
@anaradanovic Hi Ana! Let's hold off on merging this until we can chat more tomorrow :)

**Comment by nordme** [comment]
Ok, pytest is passing!!!

File: mne/preprocessing/tests/test_unify_bads.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+@pytest.mark.parametrize(
+    "instance", ("raw", "epochs", "evoked", "raw_spectrum", "epochs_spectrum")
+)
+def test_instance_support(instance, request, evoked):
+    """Tests support of different classes."""
+    # test unify_bads function on instance (single input, no bads scenario)
+    inst = _get_inst(instance, request, evoked)
+    inst_out = unify_bad_channels([inst])
+    assert inst_out == [inst]
```
Comment (drammock) [inline mne/preprocessing/tests/test_unify_bads.py]: can be removed, only need to test one instance type.

File: mne/preprocessing/tests/test_unify_bads.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    three_bad.info["bads"] = chns
+    # scenario 1: single instance passed with actual bads
+    s_out = unify_bad_channels([one_bad])
+    assert len(s_out) == 1
+    assert s_out[0].info["bads"] == [chns[1]], (s_out[0].info["bads"], chns[1])
+    # scenario 2: multiple instances passed
+    m_out = unify_bad_channels([one_bad, no_bad, three_bad])
+    assert len(m_out) == 3
+    correct_bads = [chns[1], chns[0], chns[2]]
+    for i in np.arange(len(m_out)):
```
Comment (drammock) [inline mne/preprocessing/tests/test_unify_bads.py]:
```suggestion
    for i in range(len(m_out)):
```

File: mne/channels/tests/test_unify_bads.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+from mne.channels import unify_bad_channels
+
+
+def test_error_raising(raw, epochs):
+    """Tests input checking."""
+    with pytest.raises(TypeError, match=r"must be an instance of list"):
+        unify_bad_channels("bad input")
+    with pytest.raises(ValueError, match=r"empty list"):
+        unify_bad_channels([])
+    with pytest.raises(TypeError, match=r"must be an instance of"):
```
Comment (drammock) [inline mne/channels/tests/test_unify_bads.py]:
```suggestion
    with pytest.raises(TypeError, match=r"each object in insts must be an instance of"):
```

File: mne/channels/tests/test_unify_bads.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        unify_bad_channels(["bad_instance"])
+    with pytest.raises(ValueError, match=r"same type"):
+        unify_bad_channels([raw, epochs])
+    with pytest.raises(ValueError, match=r"sorted differently"):
```
Comment (drammock) [inline mne/channels/tests/test_unify_bads.py]:
```suggestion
    with pytest.raises(ValueError, match=r"Channels do not match across"):
```

File: mne/channels/tests/test_unify_bads.py [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+        raw_alt1 = raw.copy()
+        raw_alt1.drop_channels(raw.info["ch_names"][-1])
+        unify_bad_channels([raw, raw_alt1])  # ch diff preserving order
+    with pytest.raises(ValueError, match=r"not consistent"):
```
Comment (drammock) [inline mne/channels/tests/test_unify_bads.py]:
```suggestion
    with pytest.raises(ValueError, match=r"sorted differently"):
```

File: mne/channels/tests/test_unify_bads.py [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+        raw_alt2 = raw.copy()
+        new_order = [raw.info["ch_names"][-1]] + raw.info["ch_names"][:-1]
```
Comment (drammock) [inline mne/channels/tests/test_unify_bads.py]:
```suggestion
        new_order = [raw.ch_names[-1]] + raw.ch_names[:-1]
```

File: mne/channels/tests/test_unify_bads.py [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+    Tests two cases: a) single instance passed to function with an existing
+    bad, and b) multiple instances passed to function with varying compilation
+    scenarios including empty bads, unique bads, and partially duplicated bads
+    listed out-of-order.
+
+    Only the Raw instance type is tested, since bad channel implementation is
+    controlled across instance types with a MixIn class.
+    """
+    assert raw.info["bads"] == []
+    chns = raw.info["ch_names"][0:3]
```
Comment (drammock) [inline mne/channels/tests/test_unify_bads.py]:
```suggestion
    chns = raw.ch_names[:3]
```

File: mne/channels/tests/test_unify_bads.py [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+    # scenario 1: single instance passed with actual bads
+    s_out = unify_bad_channels([one_bad])
+    assert len(s_out) == 1, len(s_out)
+    assert s_out[0].info["bads"] == [chns[1]], (s_out[0].info["bads"], chns[1])
+    # scenario 2: multiple instances passed
+    m_out = unify_bad_channels([one_bad, no_bad, three_bad])
+    assert len(m_out) == 3, len(m_out)
+    correct_bads = [chns[1], chns[0], chns[2]]
+    for i in range(len(m_out)):
+        assert m_out[i].info["bads"] == correct_bads
```
Comment (drammock) [inline mne/channels/tests/test_unify_bads.py]:
```suggestion
    expected_order = [chns[1], chns[0], chns[2]]
    for inst in m_out:
        assert inst.info["bads"] == expected_order
```

**Comment by nordme** [comment]
@anaradanovic Hi Ana! Ok, Dan thinks that we're ready to accept this PR.

**Comment by nordme** [comment]
Ok, now, actually for real we're ready to accept this PR!

**Comment by drammock** [comment]
indeed!  @anaradanovic please merge this one, then we'll do a final review/merge of the combined work

**Pull request merged**